switchtec.c

```diff
@@ -1309,23 +1310,39 @@ static int switchtec_init_pci(struct switchtec_dev *stdev,
 			      struct pci_dev *pdev)
 {
 	int rc;
+	struct resource *res;
 
 	rc = pcim_enable_device(pdev);
 	if (rc)
 		return rc;
 
-	rc = pcim_iomap_regions(pdev, 0x1, KBUILD_MODNAME);
-	if (rc)
-		return rc;
-
 	rc = dma_set_coherent_mask(&pdev->dev, DMA_BIT_MASK(64));
 	if (rc)
 		return rc;
 
 	pci_set_master(pdev);
 
-	stdev->mmio = pcim_iomap_table(pdev)[0];
-	stdev->mmio_mrpc = stdev->mmio + SWITCHTEC_GAS_MRPC_OFFSET;
+	res = devm_request_mem_region(&pdev->dev,
```

Comment:
Probably easier if you just enclose this in `if (!devm_request_mem_region(...))` seeing you don't need the resource.

Also, it will probably help the line length issues and improve readability if you store start and len in variables seeing you use the functions a lot.